// linkedin-private-api/dist/src/types/language-code.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Language = void 0;
var Language;
(function (Language) {
    Language["Afar"] = "aa";
    Language["Abkhaz"] = "ab";
    Language["Avestan"] = "ae";
    Language["Afrikaans"] = "af";
    Language["Akan"] = "ak";
    Language["Amharic"] = "am";
    Language["Aragonese"] = "an";
    Language["Arabic"] = "ar";
    Language["Assamese"] = "as";
    Language["Avaric"] = "av";
    Language["Aymara"] = "ay";
    Language["Azerbaijani"] = "az";
    Language["Bashkir"] = "ba";
    Language["Belarusian"] = "be";
    Language["Bulgarian"] = "bg";
    Language["Bihari"] = "bh";
    Language["Bislama"] = "bi";
    Language["Bambara"] = "bm";
    Language["Bengali"] = "bn";
    Language["Tibetan Standard"] = "bo";
    Language["Breton"] = "br";
    Language["Bosnian"] = "bs";
    Language["Catalan"] = "ca";
    Language["Chechen"] = "ce";
    Language["Chamorro"] = "ch";
    Language["Corsican"] = "co";
    Language["Cree"] = "cr";
    Language["Czech"] = "cs";
    Language["Old Church Slavonic"] = "cu";
    Language["Chuvash"] = "cv";
    Language["Welsh"] = "cy";
    Language["Danish"] = "da";
    Language["German"] = "de";
    Language["Divehi"] = "dv";
    Language["Dzongkha"] = "dz";
    Language["Ewe"] = "ee";
    Language["Greek"] = "el";
    Language["English"] = "en";
    Language["Esperanto"] = "eo";
    Language["Spanish"] = "es";
    Language["Estonian"] = "et";
    Language["Basque"] = "eu";
    Language["Persian"] = "fa";
    Language["Fula"] = "ff";
    Language["Finnish"] = "fi";
    Language["Fijian"] = "fj";
    Language["Faroese"] = "fo";
    Language["French"] = "fr";
    Language["Western Frisian"] = "fy";
    Language["Irish"] = "ga";
    Language["Scottish Gaelic"] = "gd";
    Language["Galician"] = "gl";
    Language["Guaran\u00ED"] = "gn";
    Language["Gujarati"] = "gu";
    Language["Manx"] = "gv";
    Language["Hausa"] = "ha";
    Language["Hebrew"] = "he";
    Language["Hindi"] = "hi";
    Language["Hiri Motu"] = "ho";
    Language["Croatian"] = "hr";
    Language["Haitian"] = "ht";
    Language["Hungarian"] = "hu";
    Language["Armenian"] = "hy";
    Language["Herero"] = "hz";
    Language["Interlingua"] = "ia";
    Language["Indonesian"] = "id";
    Language["Interlingue"] = "ie";
    Language["Igbo"] = "ig";
    Language["Nuosu"] = "ii";
    Language["Inupiaq"] = "ik";
    Language["Ido"] = "io";
    Language["Icelandic"] = "is";
    Language["Italian"] = "it";
    Language["Inuktitut"] = "iu";
    Language["Japanese"] = "ja";
    Language["Javanese"] = "jv";
    Language["Georgian"] = "ka";
    Language["Kongo"] = "kg";
    Language["Kikuyu"] = "ki";
    Language["Kwanyama"] = "kj";
    Language["Kazakh"] = "kk";
    Language["Kalaallisut"] = "kl";
    Language["Khmer"] = "km";
    Language["Kannada"] = "kn";
    Language["Korean"] = "ko";
    Language["Kanuri"] = "kr";
    Language["Kashmiri"] = "ks";
    Language["Kurdish"] = "ku";
    Language["Komi"] = "kv";
    Language["Cornish"] = "kw";
    Language["Kyrgyz"] = "ky";
    Language["Latin"] = "la";
    Language["Luxembourgish"] = "lb";
    Language["Ganda"] = "lg";
    Language["Limburgish"] = "li";
    Language["Lingala"] = "ln";
    Language["Lao"] = "lo";
    Language["Lithuanian"] = "lt";
    Language["Luba-Katanga"] = "lu";
    Language["Latvian"] = "lv";
    Language["Malagasy"] = "mg";
    Language["Marshallese"] = "mh";
    Language["M\u0101ori"] = "mi";
    Language["Macedonian"] = "mk";
    Language["Malayalam"] = "ml";
    Language["Mongolian"] = "mn";
    Language["Marathi"] = "mr";
    Language["Malay"] = "ms";
    Language["Maltese"] = "mt";
    Language["Burmese"] = "my";
    Language["Nauru"] = "na";
    Language["Norwegian Bokm\u00E5l"] = "nb";
    Language["Northern Ndebele"] = "nd";
    Language["Nepali"] = "ne";
    Language["Ndonga"] = "ng";
    Language["Dutch"] = "nl";
    Language["Norwegian Nynorsk"] = "nn";
    Language["Norwegian"] = "no";
    Language["Southern Ndebele"] = "nr";
    Language["Navajo"] = "nv";
    Language["Chichewa"] = "ny";
    Language["Occitan"] = "oc";
    Language["Ojibwe"] = "oj";
    Language["Oromo"] = "om";
    Language["Oriya"] = "or";
    Language["Ossetian"] = "os";
    Language["Panjabi"] = "pa";
    Language["P\u0101li"] = "pi";
    Language["Polish"] = "pl";
    Language["Pashto"] = "ps";
    Language["Portuguese"] = "pt";
    Language["Quechua"] = "qu";
    Language["Romansh"] = "rm";
    Language["Kirundi"] = "rn";
    Language["Romanian"] = "ro";
    Language["Russian"] = "ru";
    Language["Kinyarwanda"] = "rw";
    Language["Sanskrit"] = "sa";
    Language["Sardinian"] = "sc";
    Language["Sindhi"] = "sd";
    Language["Northern Sami"] = "se";
    Language["Sango"] = "sg";
    Language["Sinhala"] = "si";
    Language["Slovak"] = "sk";
    Language["Slovene"] = "sl";
    Language["Samoan"] = "sm";
    Language["Shona"] = "sn";
    Language["Somali"] = "so";
    Language["Albanian"] = "sq";
    Language["Serbian"] = "sr";
    Language["Swati"] = "ss";
    Language["Southern Sotho"] = "st";
    Language["Sundanese"] = "su";
    Language["Swedish"] = "sv";
    Language["Swahili"] = "sw";
    Language["Tamil"] = "ta";
    Language["Telugu"] = "te";
    Language["Tajik"] = "tg";
    Language["Thai"] = "th";
    Language["Tigrinya"] = "ti";
    Language["Turkmen"] = "tk";
    Language["Tagalog"] = "tl";
    Language["Tswana"] = "tn";
    Language["Tonga"] = "to";
    Language["Turkish"] = "tr";
    Language["Tsonga"] = "ts";
    Language["Tatar"] = "tt";
    Language["Twi"] = "tw";
    Language["Tahitian"] = "ty";
    Language["Uyghur"] = "ug";
    Language["Ukrainian"] = "uk";
    Language["Urdu"] = "ur";
    Language["Uzbek"] = "uz";
    Language["Venda"] = "ve";
    Language["Vietnamese"] = "vi";
    Language["Volap\u00FCk"] = "vo";
    Language["Walloon"] = "wa";
    Language["Wolof"] = "wo";
    Language["Xhosa"] = "xh";
    Language["Yiddish"] = "yi";
    Language["Yoruba"] = "yo";
    Language["Zhuang"] = "za";
    Language["Chinese"] = "zh";
    Language["Zulu"] = "zu";
})(Language = exports.Language || (exports.Language = {}));
