// linkedin-web-api/dist/src/requests/invitation.request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvitationRequest = void 0;
class InvitationRequest {
    constructor({ request }) {
        this.request = request;
    }
    sendInvitation({ profileId, trackingId }) {
        const requestPayload = {
            trackingId,
            emberEntityName: 'growth/invitation/norm-invitation',
            invitee: {
                'com.linkedin.voyager.growth.invitation.InviteeProfile': {
                    profileId,
                },
            },
        };
        return this.request.post('growth/normInvitations', requestPayload);
    }
    getReceivedInvitations({ skip = 0, limit = 100 } = {}) {
        const queryParams = {
            start: skip,
            count: limit,
            q: 'receivedInvitation',
        };
        return this.request.get('relationships/invitationViews', {
            params: queryParams,
        });
    }
    getSentInvitations({ skip = 0, limit = 100 } = {}) {
        const queryParams = {
            start: skip,
            count: limit,
            invitationType: 'CONNECTION',
            q: 'invitationType',
        };
        return this.request.get('relationships/sentInvitationViewsV2', {
            params: queryParams,
        });
    }
}
exports.InvitationRequest = InvitationRequest;
