// linkedin-web-api/dist/src/requests/job.request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobRequest = void 0;
class JobRequest {
    constructor({ request }) {
        this.request = request;
    }
    searchBlended({ skip = 0, limit = 10, filters = {}, keywords, }) {
        console.log('filters', filters);
        const queryParams = {
            filters,
            count: limit,
            decorationId: 'com.linkedin.voyager.deco.jserp.WebJobSearchHitLite-14',
            origin: 'JOBS_HOME_LOCATION_HISTORY',
            q: 'jserpFilters',
            queryContext: 'List(primaryHitType->JOBS,spellCorrectionEnabled->true)',
            topNRequestedFlavors: 'List(HIDDEN_GEM,IN_NETWORK,SCHOOL_RECRUIT,COMPANY_RECRUIT,SALARY,JOB_SEEKER_QUALIFIED,PREFERRED_COMMUTE,PRE_SCREENING_QUESTIONS,SKILL_ASSESSMENTS,ACTIVELY_HIRING_COMPANY,TOP_APPLICANT)',
        };
        return this.request.get('search/hits', {
            params: queryParams,
        });
    }
}
exports.JobRequest = JobRequest;
